crates/aptos-dkg/src/fiat_shamir.rs

```diff
@@ -29,25 +31,48 @@ pub const PVSS_DOM_SEP: &[u8; 21] = b"APTOS_SCRAPE_PVSS_DST"; // TODO: Name need
 /// ⚠️ This trait is intentionally private: `challenge_scalars`
 /// should **only** be used internally to ensure properly
 /// labelled scalar generation across protocols.
-trait ScalarProtocol {
-    /// **Auxiliary** function to return random scalars
-    fn challenge_scalars(&mut self, label: &[u8], num_scalars: usize) -> Vec<blstrs::Scalar>;
+pub trait ScalarProtocol<S: FromUniformBytes> {
+    fn challenge_scalars(&mut self, label: &[u8], num_scalars: usize) -> Vec<S>;
+
+    fn challenge_scalar(&mut self, label: &[u8]) -> S {
+        self.challenge_scalars(label, 1)[0]
+    }
+}
+
+pub trait FromUniformBytes: Copy {
+    const BYTE_SIZE: usize;
+    fn from_uniform_bytes(bytes: &[u8]) -> Self;
+}
+
+/// This reason for this wrapper is to avoid "overlapping implementations" for blstrs::Scalar
```

Comment:
Still don't follow: what is the "overlapping implementation" issue?

Reply:
...because `E::ScalarField` is an associated type that could be set to `blstrs::Scalar`